account_invoice_import_ubl/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2016-2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime
+
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+logger = logging.getLogger(__name__)
+
+
+class AccountInvoiceImport(models.TransientModel):
+    _name = "account.invoice.import"
+    _inherit = ["account.invoice.import", "base.ubl"]
+
+    @api.model
+    def parse_xml_invoice(self, xml_root):
+        if xml_root.tag and xml_root.tag.startswith(
+            "{urn:oasis:names:specification:ubl:schema:xsd:Invoice"
+        ):
+            return self.parse_ubl_invoice(xml_root)
+        else:
+            return super().parse_xml_invoice(xml_root)
+
+    def get_attachments(self, xml_root, namespaces):
+        attach_xpaths = xml_root.xpath(
+            "//cac:AdditionalDocumentReference", namespaces=namespaces
+        )
+        attachments = {}
+        for attach_xpath in attach_xpaths:
+            filename_xpath = attach_xpath.xpath("cbc:ID", namespaces=namespaces)
+            filename = filename_xpath and filename_xpath[0].text or False
+            data_xpath = attach_xpath.xpath(
+                "cac:Attachment/cbc:EmbeddedDocumentBinaryObject", namespaces=namespaces
+            )
+            data_base64 = data_xpath and data_xpath[0].text or False
+            if filename and data_base64:
+                if data_xpath[0].attrib and data_xpath[0].attrib.get("mimeCode"):
+                    mimetype = data_xpath[0].attrib["mimeCode"].split("/")
+                    if len(mimetype) == 2:
+                        filename = "{}.{}".format(filename, mimetype[1])
+                attachments[filename] = data_base64
+        return attachments
+
+    def parse_ubl_invoice_line(self, iline, counters, namespaces):
+        price_unit_xpath = iline.xpath(
+            "cac:Price/cbc:PriceAmount", namespaces=namespaces
+        )
+        qty_xpath = iline.xpath("cbc:InvoicedQuantity", namespaces=namespaces)
+        # Some UBL invoices don't have any InvoicedQuantity tag
+        # So we have a fallback on quantity = 1
+        qty = 1
+        uom = {}
+        if qty_xpath:
+            if float(qty_xpath[0].text):
+                qty = float(qty_xpath[0].text)
+            if qty_xpath[0].attrib and qty_xpath[0].attrib.get("unitCode"):
+                unece_uom = qty_xpath[0].attrib["unitCode"]
+                if unece_uom == "ZZ":
+                    unece_uom = "C62"
+                uom = {"unece_code": unece_uom}
+        product_dict = self.ubl_parse_product(iline, namespaces)
+        name_xpath = iline.xpath("cac:Item/cbc:Description", namespaces=namespaces)
```

Comment:
Description is not oftenly filled, name is more common (product name). I would suggest something like:
```
        if name_xpath[0].text and description_xpath[0].text:
            name = name_xpath[0].text + " / " and description_xpath[0].text
        elif name_xpath[0].text and not description_xpath[0].text:
            name = name_xpath[0].text
        else:
            name = description_xpath and description_xpath[0].text or "-"
```

Reply:
@NL66278 what do you think of this suggestion?

Reply:
Seems a logical thing todo to me. Using the maximum amount of information from the invoice.